FWCore/Framework/interface/ESProducer.h

```diff
@@ -238,11 +234,19 @@ namespace edm {
                        static_cast<const typename eventsetup::produce::product_traits<TReturn>::type*>(nullptr),
                        static_cast<const TRecord*>(nullptr),
                        iLabel);
-      consumesInfos_.push_back(std::make_unique<ESConsumesInfo>());
-      return ESConsumesCollectorT<TRecord>(consumesInfos_.back().get(), id);
+      consumesInfoPushBackNew();
+      return ESConsumesCollectorT<TRecord>(consumesInfoPointerToLastInfo(), id);
     }
 
-  private:
+    // These next five functions are intended for use in this class and
+    // class ESProducerExternalWork only. They should not be used in
+    // other classes derived from them.
+    unsigned int consumesInfoSize() const { return consumesInfos_.size(); }
+
+    void consumesInfoPushBackNew() { consumesInfos_.push_back(std::make_unique<ESConsumesInfo>()); }
```

Comment:
What if this returned the entry added? Then you wouldn't need `consumesInfoPointerToLastInfo()`.

Reply:
Done.